lib/frictionless_data_dataset.rb

```diff
@@ -214,6 +217,29 @@ def write_to_files(occurrence_ids)
             @problems << CSV::Row.new(problems_collector_header, data).to_s
           end
 
+          # Add missing_attributions.csv
+          uri = o.u_orcid.nil? ? "http://www.wikidata.org/entity/#{o.u_wikidata}" : "https://orcid.org/#{o.u_orcid}"
+          if (o.action.include?("recorded") && !o.occ_recordedByID.include?(uri)) ||
+            (o.action.include?("identified") && !o.occ_identifiedByID.include?(uri))
```

Comment:
Hmm... if e.g. the recordedByID is in GBIF, but identifiedByID isn't we print out the record to missing attributions, but it's not obvious which one is the excluded/missing ID from GBIF. 

When I did it in the javascript I split them so it's a similar structure to the agentActions extension... I got bionomia's attributions to have 1 row with id, and id_type (recordedBy or identifiedBy) (see https://github.com/gbif-norway/bionomia-roundtrip-helper/blob/master/index.html#L186) and did the same for GBIF. 

That made it easier to compare them and then it's also obvious which ID is missing. It's more work though...

Reply:
Excellent observation here. There was a tradeoff in performance to generate an output while minimizing the breadth and number of SQL calls. There's bleeding of interests. And, since you're piggy-backing off that query, it sucks. I'd not be surprised if the attributions.csv is suffering from comparable shortcomings for exactly this same reason. Am digging now to see what we're really trying to do here and find some kind of comprehensible middle ground.

Reply:
Ok so actually what I was saying isn't true, we think? 

The attributions.csv is created row by row, with either identifiedByID or recordedByID (but never both in the same row), right? And the mismatches.csv is made in exactly the same way... so mismatches.csv should also have 1 row per attribution, with either recordedByID or identifiedByID.